src/rc-update/rc-update.c

```diff
@@ -238,116 +240,96 @@ int main(int argc, char **argv)
 			return ret;
 		case_RC_COMMON_GETOPT
 		}
+	}
 
-	verbose = rc_yesno(getenv ("EINFO_VERBOSE"));
-
-	if ((action & DOSHOW   && action != DOSHOW) ||
-	    (action & DOADD    && action != DOADD) ||
-	    (action & DODELETE && action != DODELETE))
-		eerrorx("%s: cannot mix commands", applet);
-
-	/* We need to be backwards compatible */
+	verbose = rc_yesno(getenv("EINFO_VERBOSE"));
 	if (optind < argc) {
 		if (strcmp(argv[optind], "add") == 0)
-			action = DOADD;
+			action = DO_ADD;
 		else if (strcmp(argv[optind], "delete") == 0 ||
 		    strcmp(argv[optind], "del") == 0)
-			action = DODELETE;
+			action = DO_DELETE;
 		else if (strcmp(argv[optind], "show") == 0)
-			action = DOSHOW;
-		if (action)
-			optind++;
+			action = DO_SHOW;
 		else
-			eerrorx("%s: invalid command `%s'",
-			    applet, argv[optind]);
+			usage(EXIT_FAILURE);
+		optind++;
 	}
-	if (!action)
-		action = DOSHOW;
+	if (action == DO_NONE)
+		action = DO_SHOW;
 
-	runlevels = rc_stringlist_new();
+	if (optind >= argc && action != DO_SHOW)
+		usage(EXIT_FAILURE);
```

Comment:
Same here, it should still show you what you did wrong. Not just print usage with no other context.